### DISCUSSION TITLE
Which ports are used by AP systems?

### DISCUSSION THREAD

**Discussion by puttemanp** [ksheumaker/homeassistant-apsystems_ecur#249] [Q&A]
I have a solar installation in my house in Portugal. My HA instance is running in Belgium. 
However, I would like to add my portugal installation in Home Assistant. 
When I put the converter in DMZ, I can configure it in HA (I have a fixed IP in Portugal). Dmz is not what I want, and it was just a test.
An option would be a VPN, but then I'd have to renumber my network as it has the same range of one of my other networks that is already connected by VPN. 
So, I was thinking, if I know the ports that are used, I could pinhole my firewall and just open the needed ports.

**Comment by HAEdwin** [comment]
Port used is 8899 TCP (not UDP) so you can try that.

**Reply by puttemanp** [reply]
Thx! I'll try that next time I'm there, in a few weeks...

**Comment by danstowell** [comment]
According to the [Professional Installers Troubleshooting Guide](https://usa.apsystems.com/document/professional-installers-troubleshooting-guide/), the AP systems ECU uses port 21 plus many ports in the range 8995-9228. In my case I was able to verify that enabling/disabling these ranges had the corresponding effect on the ECU.

**Reply by HAEdwin** [reply]
@danstowell  You are correct, all ports mentioned are listeners **on EMA servers** so the goal of this part of the Guide is that you do not block these. If you do, data cannot be sent to EMA (and the ECU can not be updated with latest firmware for example) resulting in a internet disconnect from the ECU (lower LED on the ECU). I assume that there is an open internet connection at the remote location and the ECU is working properly there.

What @puttemanp is trying to achieve is connecting to port 8899 on the ECU from a different location, therfore the firewall on the remote location must open this port to enable the integration to query the ECU.

Note that 8899 is not officially supported by APsystems and therefor not mentioned. But nowadays a lot of Home Automation fans are using this port. Home Assitant alone has more than 700 active and happy users.

**Comment by TaylorTWBrown** [comment]
I have the new model of ECU-C. The integration works well, but the device has ports 4540 and 4560. Is anyone familiar with what these extra ports can do?

This question is not about the integration, and more about the device in general. Of course, APsystem has horrendous documentation and support, so here I am!

**Comment by HAEdwin** [comment]
@TaylorTWBrown, thank you for your question. Did you notice btw this integration has moved to [https://github.com/HAEdwin/homeassistant-apsystems_ecu_reader](https://github.com/HAEdwin/homeassistant-apsystems_ecu_reader) ? But you don't seem to be the only one who hasn't noticed this yet :)

Anyway, I already investigated this port 4540 in 2021. It contains less data than the current method, making it less appealing to investigate. But things may have changed since then. APsystems has some undocumented features that make it difficult for me to predict what they will mean for us home automation enthusiasts.

**Reply by TaylorTWBrown** [reply]
Thanks! I followed up [over there](https://github.com/HAEdwin/homeassistant-apsystems_ecu_reader/discussions/105). I know that 4540 and 8899 are well understood, but I haven't found any mention of port 4560 and I haven't had much luck prodding it
